### PR TITLE
feat(typescript): pre-load definitely typed pkg

### PR DESCRIPTION
**Pull request by bodinsamuel**
Copied a bit from #407
But kept most of the logic around "type check" similar, to avoid to many changes

### CODE REVIEW

File: src/typescript/index.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    };
+  }
+
+  if (pkg.types.ts === false) {
+    return { types: { ts: false } };
+  }
+
+  // Do we have a main .d.ts file?
+  // TO DO: replace this with a list of files check
+  if (pkg.types.ts !== 'definitely-typed' && pkg.types.ts.possible === true) {
```
Comment (Haroenv) [inline src/typescript/index.ts:82]: how can ts be definitely-typed here?

File: src/typescript/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        name: 'my-lib',
+        types: { ts: { possible: true, dtsMain: 'main.d.ts' } },
+        version: '1.0.0',
+      });
+      expect(typesSupport).toEqual({ types: { ts: 'included' } });
+    });
+  });
+
+  // TO DO : reup this
+  // adescribe('FilesList', () => {
```
Comment (Haroenv) [inline src/typescript/index.test.ts:107]: what's missing to add these tests back?

Comment (bodinsamuel) [reply src/typescript/index.test.ts:107]: In the aforementioned PR I was prefetching the list of files beforehand but it requires more work and it's not the slowest part of the code

File: src/typescript/index.ts (hunk 1 above)
Comment (bodinsamuel) [reply src/typescript/index.ts:82]: It can not but Typescript is messing with me, I prefer to have an useless if than adding `// @ts-ignore`

**Review by Haroenv — APPROVED** [review]

**Pull request merged**